src/Meshes/transformNode.ts

```diff
@@ -254,30 +252,39 @@ export class TransformNode extends Node {
      * The forward direction of that transform in world space.
      */
     public get forward(): Vector3 {
-        return Vector3.Normalize(Vector3.TransformNormal(
-            this.getScene().useRightHandedSystem ? this._forwardInverted : this._forward,
-            this.getWorldMatrix()
-        ));
+        Vector3.TransformNormalToRef(
+            this.getScene().useRightHandedSystem ? Vector3.RightHandedForwardReadOnly : Vector3.LeftHandedForwardReadOnly,
+            this.getWorldMatrix(),
+            this._forward
+        );
+        this._forward.normalize();
+        return this._forward;

```

Comment:
nit: can some some chars :-)
```suggestion
        return this._forward.normalize();
```